core/Plugin/Visualization.php

```diff
@@ -461,10 +461,30 @@ private function postDataTableLoadedFromAPI()
         if ($this->dataTable instanceof DataTable) {
             $this->metadata = $this->dataTable->getAllTableMetadata();
 
-            if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
-                $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
+
+        } else {
+            // if the dataTable is Map
+            if ($this->dataTable instanceof DataTable\Map) {
+                // load all the data
+                $dataTable = $this->dataTable->getDataTables();
+                // find the latest key
+                foreach ($dataTable as $item) {
+                    $metaData = $item->getAllTableMetadata();
+                    // if ts_created not exist exit loop
+                    if (!isset($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        break;
```

Comment:
shouldn't this be a continue, so the remaining tables will be checked for a valid metadata entry?